Core/Newsletter2Go/Bootstrap.php

```diff
@@ -253,37 +259,30 @@ public function getNewsletterGroupsApiController(Enlight_Event_EventArgs $argume
      * Event listener function of the Enlight_Controller_Dispatcher_ControllerPath_Api_NewsletterScriptUrls
      * event. This event is fired when shopware trying to access the plugin NewsletterScriptUrls controller.
      *
-     * @param Enlight_Event_EventArgs $arguments
      * @return string
      */
-    public function getNewsletterScriptUrlsApiController(Enlight_Event_EventArgs $arguments)
+    public function getNewsletterScriptUrlsApiController()
     {
         return $this->Path() . 'Controllers/Api/NewsletterScriptUrls.php';
     }
-    
+
     /**
      * Event listener function of the Enlight_Controller_Dispatcher_ControllerPath_Api_ArticleSeoLink
      * event. This event is fired when shopware trying to access the plugin ArticleSeoLink controller.
      *
-     * @param Enlight_Event_EventArgs $arguments
      * @return string
      */
-    public function getArticleSeoLinkApiController(Enlight_Event_EventArgs $arguments)
+    public function getArticleSeoLinkApiController()
     {
         return $this->Path() . 'Controllers/Api/ArticleSeoLink.php';
     }
 
     protected function registerCustomModels()
     {
-        $this->Application()->Loader()->registerNamespace(
-            'Shopware\Models\Newsletter2Go',
-            $this->Path() . 'Models/'
-        );
-        $this->Application()->ModelAnnotations()->addPaths(
-            array(
-                $this->Path() . 'Models/',
-            )
-        );
+        $dir = $this->Path();
+        $container = Shopware()->Container();
+        $container->get('loader')->registerNamespace('Shopware\Models\Newsletter2Go', $dir . 'Models/');
+        $container->get('modelannotations')->addPaths([$dir . 'Models/']);
```

Comment:
why did you used `$container->get('modelannotations')` instead of `$this->Application()->ModelAnnotations` the old approach looks much cleaner to me?

Reply:
The old approach is deprecated and will be removed in the future. See #4.

Reply:
And there is no better way for getting "loader" and "modelannotations" than by string-key?

Reply:
Nope. Internally, [this is what's being done anyway](https://github.com/shopware/shopware/blob/31d1032eee4b18b706de228bc1a79fa32ae30509/engine/Shopware/Application.php#L99) and shopware is using this mechanism [all over the place](https://github.com/shopware/shopware/search?utf8=%E2%9C%93&q=container-%3Eget%28&type=).